# Assignment1


#### Viswas Vanama

In [1]:
stopwords=['a','able','about','across','after','all','almost','also','am','among',
             'an','and','any','are','as','at','be','because','been','but','by','can',
             'cannot','could','dear','did','do','does','either','else','ever','every',
             'for','from','get','got','had','has','have','he','her','hers','him','his',
             'how','however','i','if','in','into','is','it','its','just','least','let',
             'like','likely','may','me','might','most','must','my','neither','no','nor',
           'not','of','off','often','on','only','or','other','our','own','rather','said',
             'say','says','she','should','since','so','some','than','that','the','their',
             'them','then','there','these','they','this','tis','to','too','twas','us',
             'wants','was','we','were','what','when','where','which','while','who',
             'whom','why','will','with','would','yet','you','your']



### Calculating the word frequencies for Obama tweets

In [26]:

import re
with open("obama.txt")as obama1:
#importing regular expressions ,converting file into lower
    tweets1=obama1.read().lower()
    #cleaning the tweets
    words = re.findall(r'\w+', tweets1)
    #creating an empty worddict to store words and frequencies
    worddict = {}
    for word in words: 
        if word not in stopwords and len(word)>2:
            if  word in worddict: 
                worddict[word] += 1
            else: 
                worddict[word] = 1
    #print worddict in descending order
    for w in sorted(worddict, key=worddict.get, reverse=True):
          print w, worddict[w] 

 obama 798
https 608
trump 202
president 148
miss 85
jokes 82
gonna 82
laughing 76
barack 76
biden 49
years 44
amp 42
house 41
day 40
based 39
make 38
legitimacy 38
via 37
israel 35
freddyamazin 34
h0csbsaqqc 33
lewis 33
here 31
new 30
people 30
resign 29
gop 28
king 28
45th 28
made 28
presidency 27
white 27
early 27
ruin 27
out 27
more 26
merchandise 26
timgostony 26
luther 25
call 24
video 24
martin 24
over 23
michelle 23
interview 22
country 22
skipping 22
wikileaks 22
during 21
won 21
political 21
books 21
democrats 21
racist 20
politics 20
under 19
office 19
donald 19
inauguration 19
media 19
questioned 19
land 19
mike 19
administration 19
piece 19
india 18
put 18
mlk 18
conversation 18
government 18
protection 17
broemmel 17
news 17
love 17
lie 17
manning 17
expert 17
que 17
work 16
now 16
jesseberney 16
verified 16
good 16
those 16
questions 16
world 16
legacy 15
america 15
anti 15
john 15
final 15
tortured 15
freemanning 15
abstain 14
realdonaldtrump 14
last 14
resolution 14
ad

### Calculating the word frequencies for Trump tweets same as above

In [22]:


import re
with open("trump.txt")as obama1:
    tweets1=obama1.read().lower()
    words = re.findall(r'\w+', tweets1)
    #print word
    worddict = {}
    for word in words: 
        if word not in stopwords and len(word)>2:
            if  word in worddict: 
                worddict[word] += 1
            else: 
                worddict[word] = 1
    #print worddict
    for w in sorted(worddict, key=worddict.get, reverse=True):
          print w, worddict[w]

trump 3064
https 2222
donald 375
president 248
amp 235
obama 199
inauguration 194
lewis 190
nato 185
john 144
day 143
russia 137
via 135
putin 133
realdonaldtrump 122
obsolete 113
out 110
mlk 108
people 106
germany 106
new 102
press 94
news 92
funder 90
russian 90
more 88
concerned 87
make 85
now 84
over 84
america 83
don 80
voted 78
king 78
happen 77
remark 76
dworkinreport 74
white 74
cia 73
merkel 71
democrats 71
house 69
one 68
exhibit 67
won 65
anyone 65
media 63
world 61
video 60
think 60
legitimacy 60
back 60
plan 58
elect 58
time 55
here 54
american 54
rights 54
gop 54
actually 51
http 51
black 51
martin 50
000 49
luther 49
deal 49
pay 49
know 49
trumpleaks 48
see 48
racist 48
years 47
made 47
right 46
country 45
never 45
anti 44
work 44
foxnews 44
insurance 44
foreign 44
mlkday 43
even 43
good 43
going 42
policy 42
great 42
realize 42
twitter 42
room 42
realjameswoods 42
love 41
doesn 41
before 41
fuck 41
hillary 41
civil 40
question 40
support 39
biden 39
obamacare 39
girls 3

### Manually assigning scores for words displayed above 
#### words like obama,trump,https are ignored

#### For Obama 

president=+1 148,
miss=-1 85,
jokes=-1 82,
laughing=-1 76,
legitimacy=+1 38,
israel=+1 35,
people=+1 30,
king=+1 28,
presidency=+1 27,
white=-1 27"""


#### For Trump
president=+1 248,
russia=-1 137,
putin=-1 133,
obsolete=-1 113,
people=+1 106,
germany=+1 106,
russian=-1 90,
concerned=+1 87,
over=+1 84,
america=+1 83,


### Final score
<ul>
<li>Sentiment score for <b>Obama</b> is <b>+36</b> </li>
<li>Sentiment score for <b>trump</b> is <b>+241</b></li>
</ul>